components/supervisor/frontend/src/index.ts

```diff
@@ -8,7 +8,11 @@
  * <script type="text/javascript" src="/_supervisor/frontend/main.js" charset="utf-8"></script> should be inserted to index.html as first body script,
  * all other IDE scripts should go afterwards, head element should not have scripts
  */
-
+import { IDEMetricsServiceClient, MetricsName } from "./ide/ide-metrics-service-client";
+IDEMetricsServiceClient.addCounter(MetricsName.SupervisorFrontendClientTotal).catch(() => {})
+window.onerror = (event: Event | string, source?: string, lineno?: number, colno?: number, error?: Error) => {
```

Comment:
We should use here `window.addEventListener('error', () => {....})`

otherwiwse IDE can override our onerror handler

Reply:
It is fine, we are classify them later and work with thresholds.

Let's keep it outside of this PR. Just switch to addEventListener that we don't loose data points for now.